### PR TITLE
Add ARCloud documentation

### PR DESCRIPTION
**Pull request by grozdanov**
Add ARCloud deployment instructions for:
1. Public cloud GCS and AWS
2. All-In-One/Demo (bare metal)

### CODE REVIEW

File: docs/guides/arcloud/deployments/arcloud_deployment.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+```
+
+```shell
+cd arcloud-1.7.86
+```
+
+## Configure environment
+```shell
+export NAMESPACE="arcloud"
+export DOMAIN="<Network IPv4>"
```
Comment (singlerider) [inline docs/guides/arcloud/deployments/arcloud_deployment.md:24]: "Network IPv4" will require explanation for users, as-written.

File: docs/guides/arcloud/deployments/arcloud_deployment.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+export REGISTRY_SERVER="quay.io"
+export REGISTRY_USERNAME="<username>"
+export REGISTRY_PASSWORD="<password>"
+```
+
+## Infrastructure setup
+Choose one of the following environments:
+- [Google Cloud](/docs/guides/arcloud/deployments/infrastructure/public-cloud/arcloud-deployment-gcs)
+- [AWS](/docs/guides/arcloud/deployments/infrastructure/public-cloud/arcloud-deployment-aws)
+- [All-In-One/Demo](/docs/guides/arcloud/deployments/infrastructure/bare-metal/arcloud-deployment-environment-setup)
```
Comment (singlerider) [inline docs/guides/arcloud/deployments/arcloud_deployment.md:35]: 'Preferred for this to be listed first, as it's what our current customers are trying to do.

**Review by singlerider — CHANGES_REQUESTED** [review]
Minor changes for now. I'll do another pass at another time.

File: docs/guides/arcloud/deployments/arcloud_deployment.md (hunk 2 above)
Comment (grozdanov) [reply docs/guides/arcloud/deployments/arcloud_deployment.md:35]: Re-writing of the workflow is coming.

**Comment by grozdanov** [comment]
New PR is coming with re-organized document tree structure.

**Pull request closed without merging**